zerver/lib/markdown/__init__.py

```diff
@@ -1833,13 +1833,23 @@ def handleMatch(  # type: ignore[override] # supertype incompatible with supersu
         return None, None, None
 
 
-class UserGroupMentionPattern(markdown.inlinepatterns.Pattern):
-    def handleMatch(self, m: Match[str]) -> Optional[Element]:
-        match = m.group(2)
-
+class UserGroupMentionPattern(markdown.inlinepatterns.InlineProcessor):
+    def handleMatch(  # type: ignore[override] # supertype incompatible with supersupertype
+        self, m: Match[str], data: str
+    ) -> Union[Tuple[None, None, None], Tuple[Element, int, int]]:
+        name = m.group(0)
+        match_span = m.span()
         db_data = self.md.zulip_db_data
+
+        # As patterns like @*group name* are captured.
+        start_syntax = "@*"
+        end_syntax = "*"
         if self.md.zulip_message and db_data is not None:
-            name = extract_user_group(match)
+            # As InlineProcessor API sends all the matching part
+            # coming after (?<![^\s\'\"\(,:<]), so @ is also
+            # included in name so it needs to be strip.
+            if name.startswith(start_syntax) and name.endswith(end_syntax):
```

Comment:
Should this be an `assert` rather than an `if`?

Reply:
The else for this is never expected to happen with current regex and `InlineProcessor` api. So if we do not put assert here this it will be easily spotted as a bug if any thing else ever happens.